jac-mtllm/mtllm/plugin.py

```diff
@@ -207,6 +208,24 @@ def with_llm(
         else:
             _tools = []
 
+        # FIXME: I know this is ugly, but the SemInfo, and SemRegistry doesn't
```

Comment:
There is no way to get the return type in mtllm, so I'm doing this monkey patching, we need to remvoe the SemRegistry and SemInfo and get actual proper types to make sure the types are in place.

Reply:
Ah, yes, this may need to have the type checker done first im guessing.